.changeset/soft-hairs-happen.md

```diff
@@ -0,0 +1,6 @@
+---
+"@chakra-ui/cli": minor
+---
+
+Support generating theme token type definitions from Chakra UI themes
+distributed as NPM modules
```

Comment:
````suggestion
The `tokens` command now supports generating theme token type definitions from a Chakra UI theme published as a package:

```sh
npx @chakra-ui/cli tokens <@your-org/chakra-theme-package>
```

A published theme package should export a theme object as either the `default` export or an export named `theme`.

```jsx
// chakra-theme-package/src/index.js
import { extendTheme } from '@chakra-ui/react'

const theme = extendTheme({})

// as default export
export default theme
// as named export
export { theme }
```
````

Changesets officially recommends three things that belong in a changeset:

- WHAT the change is
- WHY the change was made
- HOW a consumer should update their code

Your summary nails the WHAT, the WHY is implicit in the WHAT, and the suggestion adds the HOW for a complete picture!

Reply:
Wow, thank you so much for fleshing this out (you did a good portion of my additional documentation work for me 😝). I'm also glad you pointed out this guidance from the **Changesets** documentation, as I'm definitely going to be calling this out in the repositories I manage with Changesets at work!

Reply:
No problem! It comes pretty naturally at this point having written so many! Also I'm working on building a documentation site for `changesets` so that kinda helps too 😆 

Glad to hear it'll help at work too!